### PR TITLE
Allow building a static lib

### PR DESCRIPTION
**Pull request by enen92**
Hey @narugit, thanks for merging the gpu PR.
This is my last PR on the queue. I understand if you don't think it is of any value since it's pretty external to the tool (since it is a cli tool/executable). Regardless, I think it's better to push it upstream instead of carrying the patch around (and it is really small anyway).

It allows to build the project (except the executable) as a static library so that it can be included/linked in external software. With this I can include it as a dependency on Kodi and replace our old smc code (which I plan to do this week - have a small PoC that needs cleanup).

I'd ask you to please tag a new version/release after this one is merged (or refused) since it is easier for us to control the source tarballs on the mirrors.

Thanks for all your help with the recent PRs

### CODE REVIEW

**Comment by enen92** [comment]
Kodi PoC: https://github.com/xbmc/xbmc/commit/35729b97aa7e1f2868a059f7810fea67777dad0e
(still using an old version of one of my branches)

File: Makefile [hunk 1]
```diff
@@ -18,10 +21,22 @@ else
 	$(error Not support architecture: $(ARCH))
 endif
 
+OBJS= smctemp.o \
```
Comment (narugit) [inline Makefile:24]:
```suggestion
OBJS := smctemp.o \
```

File: Makefile [hunk 2]
```diff
@@ -18,10 +21,22 @@ else
...
+      smctemp_string.o
+
+HEADERS = smctemp.h \
```
Comment (narugit) [inline Makefile:27]:
```suggestion
HEADERS := smctemp.h \
```

File: Makefile [hunk 3]
```diff
@@ -1,7 +1,10 @@
 CXX = g++
 CXXFLAGS = -Wall -std=c++17 -g -framework IOKit
 EXES = smctemp
-DEST = /usr/local/bin
+LIB = libsmctemp.a
```
Comment (narugit) [inline Makefile:4]:
```suggestion
STATIC_LIB := libsmctemp.a
```

**Review by narugit — CHANGES_REQUESTED** [review]
Thank you for the suggestion. I'm in favor of adding the option to generate a static library to make it easier to port to other modules.

I haven't been differentiating between `:=` and `=` in my Makefiles so far, but I'm considering a reevaluation as part of this PR. To ensure a reliable build with this Makefile, I'm leaning towards using `:=` exclusively unless there's a spetial reason not to. In the previous Makefiles, even `CXX` was assigned using `=`, but I plan to switch it to `:=` after this PR.

**Review by narugit — APPROVED** [review]
LGTM

**Comment by narugit** [comment]
@enen92 
I've created 0.2.0 release.
https://github.com/narugit/smctemp/releases/tag/0.2.0

**Pull request merged**